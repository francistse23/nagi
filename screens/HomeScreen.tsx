import React from "react";
import { Image, StyleSheet, SafeAreaView } from "react-native";
import { useNavigation } from "@react-navigation/native";
import { LinearGradient } from "expo-linear-gradient";

import { Button, LargeText, MediumText } from "../styled-components";
import Constants from "../constants";
import { scale, verticalScale } from "../utilities/scale";

const { mainColor, secondaryColor } = Constants;

export default function HomeScreen(): React.ReactElement {
  const { navigate } = useNavigation();

  return (
    <SafeAreaView style={styles.container}>
      <LinearGradient
        colors={[secondaryColor, mainColor]}
        end={[0, 0.7]}
        style={{
          alignItems: "center",
          flex: 1,
          justifyContent: "space-between",
          paddingVertical: "10%",
          width: "100%",
        }}
      >
        <LargeText
          style={{
            fontSize: scale(60),
          }}
        >
          Nagi
        </LargeText>
        <Image
          source={require("../assets/nagi.png")}
          style={{
            height: verticalScale(197),
            marginVertical: verticalScale(50),
            width: scale(200),
          }}
        />
        <MediumText style={{ width: "60%" }}>
          "Quiet the mind, and the soul will speak."
        </MediumText>
        <Button onPress={() => navigate("Meditation")}>
          <MediumText>Start Meditating</MediumText>
        </Button>
      </LinearGradient>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    alignItems: "center",
    flex: 1,
    justifyContent: "center",
  },
});
