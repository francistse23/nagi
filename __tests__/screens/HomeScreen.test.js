import React from "react";
import renderer from "react-test-renderer";

import HomeScreen from "../../screens/HomeScreen";

describe("Home screen renders correctly", () => {
  it("renders correctly", () => {
    const screen = renderer.create(HomeScreen);
    expect(screen).toMatchSnapshot();
  });
});
