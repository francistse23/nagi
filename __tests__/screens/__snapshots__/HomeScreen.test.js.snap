// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home screen renders correctly renders correctly 1`] = `null`;
